Source/swiftlint/Extensions/LinterCache+CommandLine.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  LinterCache+CommandLine.swift
+//  SwiftLint
+//
+//  Created by Marcelo Fabri on 12/31/16.
+//  Copyright © 2016 Realm. All rights reserved.
+//
+
+import Foundation
+import SwiftLintFramework
+
+extension LinterCache {
+
+    class func makeCache(options: LintOptions, configuration: Configuration) -> LinterCache? {
+        guard let url = cacheUrl(options: options, configuration: configuration) else {
+            return nil
+        }
+
+        let configurationHash = configuration.hash
+        let cache: LinterCache
+        do {
+            cache = try LinterCache(contentsOf: url, configurationHash: configurationHash)
+        } catch {
+            cache = LinterCache(configurationHash: configurationHash)
+        }
+
+        return cache
+    }
+
+    func save(options: LintOptions, configuration: Configuration) {
+        if let url = cacheUrl(options: options, configuration: configuration) {
+            try? save(to: url)
+        }
+    }
+
+}
+
+private func cacheUrl(options: LintOptions, configuration: Configuration) -> URL? {
+    guard !options.ignoreCache else {
+        return nil
+    }
+    let path = (options.cachePath.isEmpty ? configuration.cachePath : options.cachePath) ?? ".swiftlint_cache.json"
```

Comment:
Do you have any ideas on how to do this? I've thought about several ways, but each one of them have big disadvantages:

1. Keep an index file where we map root path to a cache file named from an UUID. I think this is my favorite option so far.
2. Use MD5 to generate the cache file name, but it's not easy to do it in Swift.
3. Use `.hashValue` on the root path name to generate the cache name, but I don't know how much we can trust on it.

We also need to have different defaults for Linux and Darwin, since Linux doesn't have `~/Library/Application Support`. It doesn't even implement `FileManager.default.urls(for: .applicationSupportDirectory, in: .userDomainMask)`.